src/client/puppet-service.ts

```diff
@@ -1449,13 +1450,13 @@ export class PuppetService extends Puppet {
 
   public async friendshipAdd (
     contactId : string,
-    hello     : string,
+    options   : FriendshipAddOptions,
   ): Promise<void> {
-    log.verbose('PuppetService', 'friendshipAdd(%s, %s)', contactId, hello)
+    log.verbose('PuppetService', 'friendshipAdd(%s, %s)', contactId, JSON.stringify(options))
 
     const request = new FriendshipAddRequest()
     request.setContactId(contactId)
-    request.setHello(hello)
```

Comment:
For backward compatibility, it would be better to also send hello in the request.